app_users/models.py

```diff
@@ -236,6 +236,13 @@ def get_or_create_personal_workspace(self) -> tuple["Workspace", bool]:
     def get_anonymous_token(self):
         return auth.create_custom_token(self.uid).decode()
 
+    def get_workspaces(self):
+        from workspaces.models import Workspace
+
+        return Workspace.objects.filter(
+            memberships__user=self, memberships__deleted__isnull=True
+        )
```

Comment:
a suggestion I have here - to better cache the memberships too (for access control) - is

```python
from django.utils.functional import cached_property

@cached_property
def workspace_memberships(self):
    return self.workspacemembership_set.select_related("workspace")

def get_workspaces(self):
    return [m.workspace for m in self.workspace_memberships]
```

`cached_property` has a similar usage example in the django documentation - https://docs.djangoproject.com/en/5.1/ref/utils/#module-django.utils.functional

Reply:
This will save trips to the database when we
- render the global workspace selector
- render publish dialog
- check access control / render workspace author for examples
- access control & ownership when user submits a run

Reply:
looks like we only use it in one place right now so seems un-necessary but yeah lets add it if we need membership data access in the future